### DISCUSSION TITLE
About smagorinsky_sfs_term

### DISCUSSION THREAD

**Discussion by bronise777** [AMReX-Combustion/PeleC#848] [General]
In the _**pc_smagorinsky_sfs_term**_ function, **_sigma_** is solved in different directions.
  const amrex::Real sigmadx =  Cs2 * alphaij[0] - third * CI * alpha * static_cast<amrex::Real>(dir == 0);
  const amrex::Real sigmady = Cs2 * alphaij[1] - third * CI * alpha * static_cast<amrex::Real>(dir == 1);
  const amrex::Real sigmadz = Cs2 * alphaij[2] - third * CI * alpha * static_cast<amrex::Real>(dir == 2);
But in the _**pc_dynamic_smagorinsky_sfs_term**_ function, **_sigma_** is a little different, why subtract **_CI*alpha/ 3_** only in the x direction?
  const amrex::Real sigmadx = coeff(iv, comp_Cs2) * alphaij(iv, 0) - coeff(iv, comp_CI) * alpha(iv) / 3.0;
  const amrex::Real sigmady = coeff(iv, comp_Cs2) * alphaij(iv, 1);
  const amrex::Real sigmadz = coeff(iv, comp_Cs2) * alphaij(iv, 2);

**Comment by marchdf** [comment]
I think you have indeed identified another bug. Thank you for taking the time to go through this code so carefully and raising issues when you encounter them. It is _much_ appreciated. As you can tell, the dynamic Smagorinsky LES has not received much attention since it was implemented more than 5 years ago. I hope this is the last of them. PR created for this issue in #849

**Reply by bronise777** [reply]
Have you calculated with the corrected model? In the case of my Taylor-Green vortex, the corrected calculation appears to be less good than the pre-modified one. The following figures are the results before and after correction respectively, using the case of PeleLMeX script drawing pictures
![图片1](https://github.com/user-attachments/assets/9e49d126-518b-4534-a6d8-48ed772df00f) before
![图片2](https://github.com/user-attachments/assets/e7d8421b-0c3c-4c77-b713-b54bf5c715cf)  after

**Reply by marchdf** [reply]
(Just a note, you are using PeleC, right? your labels and messages indicate PeleLMeX but this is the PeleC repo)

Something does look strange. I pored over the code again and I do think you did indeed find a bug (the one that is fixed). It is possible that, in fixing the bug, we made an other more subtle bug more obvious. I remember now that my main issue when implementing all this was that I couldn't figure out a good unit test for this. I guess we are now facing this issue. Since you've been pouring over the equations much more recently than I have, maybe you could help out with this? In an ideal world, we could set up a known initial condition and compute the expected sfs term manually/some python script and compare to what we get out of PeleC.
